spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1703,33 +1710,42 @@ public Boolean isMandatoryFor(final Message message) {
 
 	private <T> T doExecute(ChannelCallback<T> action, ConnectionFactory connectionFactory) {
 		Assert.notNull(action, "Callback object must not be null");
-		Channel channel;
+		Channel channel = null;
+		boolean invokeScope = false;
+		if (this.activeTemplateCallbacks.get() > 0) {
+			channel = this.dedicatedChannels.get();
```

Comment:
Looks like we don't need this `this.activeTemplateCallbacks`, the simple `this.dedicatedChannels.get()` works here the same way. See your logic there in the `invoke()`:
```
Assert.state(currentChannel == null, () -> "Nested invoke() calls are not supported; channel '" + currentChannel
				+ "' is already associated with this thread");
```

I won't mind if I am missing something on the matter...